### PR TITLE
[Wasm]: Change CMake logic

### LINKED ISSUE #2370: Improve cmake usage in Wasm build script
Currently we overwrite the main `CMakeLists.txt` which is not the best practice, @Milerius suggests to use cmake toolchain

### PR DESCRIPTION
**Pull request by Milerius**
## Description

- Use toolchain
- Use option for wasm
- Use make/cmake instead of emmake/emcmake
- fix #2370

## How to test

wasm-build + npm tests

## Types of changes

* Improvements

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: tools/wasm-build [hunk 1]
```diff
@@ -13,12 +13,10 @@ build_folder=${src_dir}/wasm-build
 boost_dir=${EMSDK}/upstream/emscripten/system/include
 pushd ${src_dir}
 
-cp cmake/Wasm.cmake CMakeLists.txt
-
 # cmake
-emcmake cmake -Bwasm-build -DBoost_INCLUDE_DIR=${boost_dir}
+cmake -Bwasm-build -DBoost_INCLUDE_DIR=${boost_dir} -DTrustWalletCore_USE_WASM=ON -DCMAKE_TOOLCHAIN_FILE=${EMSDK}/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake
```
Comment (hewigovens) [inline tools/wasm-build:17]: Can we omit `TrustWalletCore` prefix? `TrustWalletCore_USE_WASM` looks long and unnecessary

File: cmake/StandardSettings.cmake [hunk 2]
```diff
@@ -3,4 +3,10 @@
...
 
-option(${PROJECT_NAME}_ENABLE_CLANG_TIDY "Enable static analysis with Clang-Tidy." OFF)
\ No newline at end of file
+option(${PROJECT_NAME}_ENABLE_CLANG_TIDY "Enable static analysis with Clang-Tidy." OFF)
+
+#
+# Specific platforms support
+#
+# Currently supporting: Wasm.
+option(${PROJECT_NAME}_USE_WASM "Target Wasm" OFF)
```
Comment (hewigovens) [inline cmake/StandardSettings.cmake:12]:
```suggestion
option(${PROJECT_NAME}_COMPILE_WASM "Target Wasm" OFF)
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -112,7 +116,7 @@ target_include_directories(TrustWalletCore
         ${CMAKE_CURRENT_SOURCE_DIR}/build/local/include
 )
 
-if(NOT ANDROID AND NOT IOS_PLATFORM)
+if(NOT ANDROID AND NOT IOS_PLATFORM AND NOT ${PROJECT_NAME}_USE_WASM)
     add_subdirectory(tests)
     add_subdirectory(walletconsole/lib)
     add_subdirectory(walletconsole)
```
Comment (hewigovens) [inline CMakeLists.txt:122]: Let's add another flag to compile `walletconsole`? and make it default on CI

File: tools/wasm-build (hunk 1 above)
Comment (Milerius) [reply tools/wasm-build:17]: Yes, it's just that it use `PROJECT_NAME` internally

File: CMakeLists.txt (hunk 3 above)
Comment (Milerius) [reply CMakeLists.txt:122]: > Let's add another flag to compile `walletconsole`? and make it default on CI

In Linux CI?

File: tools/wasm-build (hunk 1 above)
Comment (hewigovens) [reply tools/wasm-build]: Yes, `TW` looks good

File: CMakeLists.txt (hunk 3 above)
Comment (hewigovens) [reply CMakeLists.txt:122]: Modular build targets, more readable

**Review by hewigovens — APPROVED** [review]

**Review by hewigovens — APPROVED** [review]

File: tools/wasm-build [hunk 4]
```diff
@@ -13,12 +13,10 @@ build_folder=${src_dir}/wasm-build
...
 
-cp cmake/Wasm.cmake CMakeLists.txt
-
 # cmake
-emcmake cmake -Bwasm-build -DBoost_INCLUDE_DIR=${boost_dir}
+cmake -Bwasm-build -DBoost_INCLUDE_DIR=${boost_dir} -DTW_COMPILE_WASM=ON -DCMAKE_TOOLCHAIN_FILE=${EMSDK}/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake
 
 # make
-emmake make -Cwasm-build
+make -j8 -Cwasm-build
```
Comment (miloserdow) [inline tools/wasm-build:20]: Maybe we should define -j8 in some variable (e.g. MAKEOPTS) instead to improve readability?

Comment (Milerius) [reply tools/wasm-build:20]: I think this makes sense in Makefile - however, I'm planning to move most of the tools to CMake targets and use ninja

**Pull request merged**